File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        n = len(nums)
        if not n:
            return False
        nums.sort()
        prev_elem = nums[0]
        for i in range(1, n):
            if nums[i] == prev_elem:
                return True
            prev_elem = nums[i]
        return False
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let res = new Map()
        for (let i of nums)
            res.set(i, (res.get(i) || 0) + 1)
        let sorted_keys = [... res.entries()].sort((a, b) => b[1] - a[1]).map(entry => entry[0])
        return sorted_keys.slice(0, k)
    }
}
